### PR TITLE
initial pass for twitter auth

### PR DESCRIPTION
**Pull request by benny-conn**
Changes:

- **Added** a new `external_socials` jsonb column to the users pii table
- **Added** a new `social_account_auth` table for storing access tokens and the like, currently supports the OAuth pattern can be extended to support different types of tokens in the future
- **Added** resolvers for connecting social accounts and resolving them
- **Added** a `socialauth` package for various social media authenticator implementations
- **Added** a `twitter` package for making calls to the twitter API (including authenticating with the API)
- **Added** a `SocialsAPI` to the `PublicAPI` for interacting with the various socials, for now just for creating the twitter authenticator

### CODE REVIEW

**Review by Robinnnnn — COMMENTED** [review]
thanks for crushing this benny! do you think we could include some basic tests?

File: service/twitter/twitter.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		err = util.GetErrFromResp(accessResp)
+		return AccessTokenResponse{}, fmt.Errorf("failed to get access token, returned status: %s", err)
+	}
+
+	var accessToken AccessTokenResponse
+	if err := json.NewDecoder(accessResp.Body).Decode(&accessToken); err != nil {
+		return AccessTokenResponse{}, err
+	}
+
+	err = a.queries.UpsertSocialMediaOAuth(ctx, coredb.UpsertSocialMediaOAuthParams{
```
Comment (radazen) [inline service/twitter/twitter.go:109]: Is this the only place we use the `queries` object in here? It feels a little deceptive for a function called `get...` to write to the database. If this is the only database interaction in the Twitter service, maybe we can move it to the calling method (something in the public API?) and write the token info there?

File: db/migrations/core/000059_social_accounts.up.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    last_updated timestamp with time zone default CURRENT_TIMESTAMP not null,
+    user_id varchar(255) not null references users(id),
+    provider varchar not null,
+    access_token varchar,
+    refresh_token varchar
+);
+
+grant select on pii.social_account_auth to access_ro_pii;
+grant all on pii.social_account_auth to access_rw_pii;
+alter table pii.social_account_auth owner to access_rw_pii;
```
Comment (radazen) [inline db/migrations/core/000059_social_accounts.up.sql:24]: We should be able to omit the grant/alter statements here; everything gets handled automatically as long as we `set role to access_rw_pii;` (and we'd get a permission denied error if we didn't!)

**Review by radazen — COMMENTED** [review]
🔥 🔥 🔥

**Review by radazen — APPROVED** [review]
🙌🙌🙌

Awesome! Looking great, Benny. Thanks for following up on comments so quickly, too!

**Pull request merged**